cmd/restic/cmd_dump.go

```diff
@@ -90,13 +90,31 @@ func printFromTree(ctx context.Context, tree *restic.Tree, repo restic.Repositor
 				return printFromTree(ctx, subtree, repo, item, pathComponents[1:], pathToPrint)
 			case node.Type == "dir":
 				node.Path = pathToPrint
-				return tarTree(ctx, repo, node, pathToPrint)
+				tw := tar.NewWriter(os.Stdout)
+				defer tw.Close()
+				return tarTree(ctx, repo, node, pathToPrint, tw)
 			case l > 1:
 				return fmt.Errorf("%q should be a dir, but s a %q", item, node.Type)
 			case node.Type != "file":
 				return fmt.Errorf("%q should be a file, but is a %q", item, node.Type)
 			}
 		}
+
+		// If we print / we need to asume that there are multiple nodes at that
+		// level in the tree.
+		if pathToPrint == "/" {
+			tw := tar.NewWriter(os.Stdout)
+			defer tw.Close()
```

Comment:
`Close()` returns an error which isn't checked here, but needs to be. I'd remove this line and handle the case explicitly as shown below. I'd also not use `defer()`, because error checking in `defer` is very error prone.
```suggestion
```